api/src/main/java/org/openmrs/liquibase/ChangeLogDetective.java

```diff
@@ -162,8 +162,7 @@ public String getInitialLiquibaseSnapshotVersion(String context, LiquibaseProvid
 	}
 	
 	List<String> getSnapshotVersionsInDescendingOrder(Map<String, List<String>> snapshotCombinations) {
-		List<String> versions = new ArrayList<>();
-		versions.addAll(snapshotCombinations.keySet());
+		List<String> versions = new ArrayList<>(snapshotCombinations.keySet());
```

Comment:
helo @sacull,thanks for this.  does this line handle the addAll() method?

Reply:
@HerbertYiga Functionally it is the same. The collection given in the constructor is added to the new collection just like addAll to empty. From our point of view, the only difference is the reduction in the number of lines of code. Internally, one does not call the other, but they use the same System.arraycopy mechanisms.

Reply:
@sacull   point noted ,thanks  cc @dkayiwa  @ibacher